js/plugins/drupalentity/plugin.js

```diff
@@ -8,27 +8,25 @@
   "use strict";
 
   CKEDITOR.plugins.add('drupalentity', {
-    // This plugin requires the Widgets System defined in the 'widget' plugin.
     requires: 'widget',
 
-    // The plugin initialization logic goes inside this method.
     beforeInit: function (editor) {
       // Configure CKEditor DTD for custom drupal-entity element.
       // @see https://www.drupal.org/node/2448449#comment-9717735
-      var dtd = CKEDITOR.dtd, tagName;
-      dtd['drupal-entity'] = {'#': 1};
-      // Register drupal-entity element as allowed child, in each tag that can
-      // contain a div element.
-      for (tagName in dtd) {
-        if (dtd[tagName].div) {
-          dtd[tagName]['drupal-entity'] = 1;
-        }
-      }
-
+      CKEDITOR.dtd['drupal-entity'] = {'#': 1};
```

Comment:
As far as comments in Drupa tickets go, you also might want to nest `drupal-entity` elements, so you also need to change this line to:
`CKEDITOR.dtd[ 'drupal-entity' ] = { '#': 1, 'drupal-entity': 1 };`